### ISSUE TITLE
Need proper handling on bridge port trunk/access mode changing

### ISSUE THREAD

**Issue by cathay4t** [nmstate/nmstate#1080]
When user switching bridge port between trunk and access mode, should we:
 * Merge `tag` and other options from current and remove incompatibly options.
 * User need to define the whole configuration for this port again when switching mode.

**Comment by cathay4t** [comment]
@maiqueb Suggest we do the same with bond mode change: user need to explicit when switching port mode with whole configurations.

**Comment by maiqueb** [comment]
> @maiqueb Suggest we do the same with bond mode change: user need to explicit when switching port mode with whole configurations.

Right. I think that keeping interface state merging on scenarios such as port mode update will lead to surprises on the user base (e.g. update trunk port to access port fails; update trunk port w/ native vlan to access port succeeds is an example), which will ultimately confuse nmstate users.

While losing some flexibility, I think forcing the user to specify the whole port configuration on port mode changes makes the user experience more predictable.
